LibrariesHacked/catalogues-library
# Versioning, Tests, fixes and updates

A mixed bag of fixes and changes since many of the tests were failing since they were originally written. **All tests now pass**.

# Versioning

Arena and Koha now have versioned implementation classes. This has reduced the amount of config required for each service in `data.json` (e.g. portlet vs portlets for Arena, etc). Version numbers of the library systems have been discovered through HTTP headers or HTML `<head>` information.

I'm surmising - though don't know for sure - that these two platforms are self-hosted and that will explain the variance in deployed versions, whereas others are SaaS products which will generally be kept up-to-date.

# Library listings

No libraries have hard-coded values anymore; all are discovered using code. 

The list of common "non-library" terms has also been updated to add some new values that have popped up since the code was originally written. 

It also now requires an exact (but case-insensitive) match on the trimmed value since it was removing some items as false-positives previously.

# Prism

Change in domain name here - the old Capita Discovery one is no longer maintained and has an expired SSL certificate.

# Tests

Tests now:

* check that a valid array of libraries is returned from `getLibraries` calls;
* ensure that availability information is valid (i.e. not "0 of 0" available).

Fixes were made where tests failed (which is where the majority of changes in this PR originate from).

# Libraries changing platform

Some libraries have switched between providers and `data.json` has been updated accordingly:

* Hartlepool has moved from Prism to Spydus
* Staffordshire has moved from Arena to Enterprise
